### ISSUE TITLE
Add coloured deep diffs

### ISSUE THREAD

**Issue by alexharv074** [Sceptre/sceptre#1279]
### Subject of the issue
See #1261 

### Steps to reproduce

A deep diff like this one also should be coloured:
<img width="1440" alt="screen_shot_2022-11-22_at_1 54 32_pm" src="https://user-images.githubusercontent.com/2305330/209462075-09f0baba-3da5-4a82-a162-390d2db08fbf.png">

**Comment by alexharv074** [comment]
An implementation similar to #1261  should work but I will assume a line beginning with 12 spaces then a minur or plus gets coloured.

**Comment by jfalkenstein** [comment]
Interesting. You'd need to have a recursive function that finds keys with the name of `diff`. It'd be a bit tricky, but possible.

**Comment by alexharv074** [comment]
@jfalkenstein  I am stumped. Your comment in the code mentions: 
> PyYaml disallows block styling for multiline strings if any of the lines has a space followed
>        by a line break.
I have found that if I insert any Fore colour code into the string the block styling is broken. Do you recall where you found the details about trailing spaces, or do you have any other ideas for how this could be done?

**Comment by jfalkenstein** [comment]
I believe I ran into this behavior through testing and following the debugger deep, deep into PyYaml code. Why? Is this proving a problem for you?

**Comment by alexharv074** [comment]
@jfalkenstein  Yes, it's proving to be a problem. Without delving as far into the internals of PyYaml, I think it is almost impossible. Any insertion of a colour code into the string causes PyYaml to not use the block styling.

**Comment by jfalkenstein** [comment]
Yeah, I remember that being an ugly dungeon of deep recursion...

**Comment by jfalkenstein** [comment]
HOWEVER, I think rather than doing that, I think it might be a better idea to apply the coloring AFTER the code comes out of py-yaml.
